src/libraries/System.IO.Hashing/src/System/IO/Hashing/Crc32.cs

```diff
@@ -169,6 +169,16 @@ public static int Hash(ReadOnlySpan<byte> source, Span<byte> destination)
 
         private static uint Update(uint crc, ReadOnlySpan<byte> source)
         {
+#if NET7_0_OR_GREATER
```

Comment:
"Deconstruct" the `source` here into
```c#
ref byte src = ref MemoryMarshal.GetReference(source);
nint length = source.Length;
```
and use these values throughout the rest of the method.